src/simulator/lidar.hh

```diff
@@ -0,0 +1,90 @@
+#ifndef LIDAR_HH
+#define LIDAR_HH
+
+#include "logging.hh"
+#include "utils.hh"
+
+#include <fmt/chrono.h>
+#include <nova/json.h>
+#include <nova/utils.h>
+#include <nova/vec.h>
+
+using json = nova::json;
+
+
+class lidar {
+public:
+    lidar(const json& config)
+        : m_config(config)
+    {
+        m_ang_res_h = m_config.lookup<float>("rpm.value") / 60 * 360 * m_config.lookup<float>("firing_cycle");
+        m_angles_hor = nova::linspace(nova::range{ 0.f, m_config.lookup<float>("fov_h") }, static_cast<std::size_t>(m_config.lookup<float>("fov_h") / m_ang_res_h), false);
+        m_angles_ver = nova::linspace(nova::range{ -m_config.lookup<float>("fov_v") / 2, m_config.lookup<float>("fov_v") / 2 }, m_config.lookup<std::size_t>("channels"), true);
```

Comment:
Consider member initializer list.

Reply:
Also, it might be worth to refactor into small functions and split the long linges.

Reply:
Somewhat done